test/e2e/serial/tests/tolerations.go

```diff
@@ -350,6 +351,83 @@ var _ = Describe("[serial][disruptive][slow][rtetols] numaresources RTE tolerati
 					g.Expect(int(updatedDs.Status.NumberReady)).To(Equal(len(pods)), "updated DS ready=%v expected pods", updatedDs.Status.NumberReady, len(pods))
 				}).WithTimeout(5 * time.Minute).WithPolling(10 * time.Second).Should(Succeed())
 			})
+
+			It("[tier3][test_id:72859] should not restart a running RTE pod on tainted node with NoSchedule effect", func(ctx context.Context) {
+				By("taint one worker node")
+				var err error
+				workers, err = nodes.GetWorkerNodes(fxt.Client, ctx)
+				Expect(err).ToNot(HaveOccurred())
+
+				tnts, _, err = taints.ParseTaints([]string{testTaint()})
+				Expect(err).ToNot(HaveOccurred())
+
+				tnt := &tnts[0]
+				taintedNode := &workers[0]
+				Eventually(func() error {
+					var err error
+					node := &corev1.Node{}
+					err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(taintedNode), node)
+					if err != nil {
+						return err
+					}
+
+					updatedNode, updated, err := taints.AddOrUpdateTaint(node, tnt)
+					if err != nil {
+						return err
+					}
+					if !updated {
+						return nil
+					}
+
+					klog.Infof("adding taint: %q to node: %q", tnt.String(), updatedNode.Name)
+					err = fxt.Client.Update(ctx, updatedNode)
+					if err != nil {
+						return err
+					}
+					return nil
+				}).WithPolling(1 * time.Second).WithTimeout(1 * time.Minute).ShouldNot(HaveOccurred())
```

Comment:
please move to `applyTaintToNode` (or let's talk why is not suitable)

Reply:
thanks for pointing out, I missed that.